src/audio/pcm_converter/pcm_converter_hifi3.c

```diff
@@ -658,12 +658,67 @@ static void pcm_convert_s16_to_f_lin(const void *psrc, void *pdst,
 	}
 }
 
+#if !defined(XT_ROUND_S)
+/**
+ * shift d value left (for positive a) or right (for negative a)
+ * and take care about overflows
+ */
+static inline int32_t _pcm_shift(int32_t d, int32_t a)
+{
+	int64_t dd = d;
+
+	if (a > 32)
+		a = 32;
+	else if (a < -32)
+		a = -32;
+
+	dd = a >= 0 ? dd << a : dd >> -a;
+	if (dd > INT32_MAX)
```

Comment:
Argument d is signed type so should you also check for INT32_MIN? If it's always non-negative, please add a comment about that.

Reply:
function ：_pcm_convert_f_to_i	This is called here, so it is a positive number . I will add a comment
```
static int32_t _pcm_convert_f_to_i(int32_t src, int32_t pow)
{
	int32_t exponent;
	int32_t mantissa;
	int32_t dst;

	exponent = (src >> 23);
	exponent = (exponent & 0xFF) + pow - 127; /* exponential */
	mantissa = BIT(23) | (MASK(22, 0) & src); /* mantissa + 1.0 [Q9.22] */
	/* calculate power */
	dst = _pcm_shift(mantissa, exponent - 23);
	/* add 0.5 to round correctly but assert it doesn't lead to overflow */
	if (exponent - 22 < 9 || (src & BIT(31)) == BIT(31))
		dst += _pcm_shift(mantissa, exponent - 22) & 1;
	/* copy sign to dst */
	dst = (dst ^ (src >> 31)) + (int)((unsigned int)src >> 31);

	return dst;
}
```